### ISSUE TITLE
Write internal ResourceFile representation to disk.

### ISSUE THREAD

**Issue by tjhancocks** [tjhancocks/kestrel-development-kit#2]
**Is your feature request related to a problem? Please describe.**
The `kdk::target` class is currently building a representation of the final resource file by adding resources to an `rsrc::file`. However this is not written to disk, and simply discarded as the assembler exits.

**Describe the solution you'd like**
A method (or set of methods) need to be introduced that allow writing the contents of the `rsrc::file` object to disk, in the binary format of either a Standard ResourceFork or the custom Extended ResourceFork.

**Comment by tjhancocks** [comment]
The implementation for the older standard resource files is completed.

**Comment by andrews05** [comment]
Are you planning to support the windows .rez format?

**Comment by tjhancocks** [comment]
This could certainly be added!

Ultimately we'd need to have a flag/option on the assembler to define what the output format should be. Something like:

```sh
--format=kestrel
--format=nova-mac
--format=nova-rez
```

I do have some old code lying around somewhere that should contain the general rez format. I can try and get that posted up on gist, for whomever wants to implement that.

**Comment by andrews05** [comment]
Yeah, I've got the format around, could possibly look at implementing it.
Given .rez doesn't have the file size limitation that standard resourcefork does, what is the advantage of the extended format? I don't think other limitations like 16-bit ids have ever been a concern.

[edit] [Here's my plugconvert code](https://github.com/andrews05/evstuff/blob/master/sources/plugconvert.c) which can write .rez files.

**Comment by tjhancocks** [comment]
To be honest the extended one came out of wanting to be able to extend the format with new features if and when required. Given that Kestrel will be the only consumer of such a format, it can be extended and tweaked right up to the point of Kestrel's release. In theory the format can also be versioned (using the version field in its header), so it can be extended later as well.

Now whether these extensions ever happen or not is another discussion.

I also don't want to tie Kestrel to the older ResourceFile and Rez files that Nova used, and allow it to be able to evolve beyond. I've mentioned in a few places that the initial goal for Kestrel is to allow cloning the _EV Nova_ engine, but beyond the _EV: Override Remastering_ it will be its own thing, with additional features, capabilities and such.

**Comment by andrews05** [comment]
Awesome, sounds good. I'm sure we've all got plenty of ideas on how it could be extended/enhanced :)
